### PR TITLE
chore(dot/rpc): Load test submit extrinsic

### LINKED ISSUE #2011: Transaction Load-Testing for Gossamer Devnet
## Task summary

- Research and document a strategy for load-testing Gossamer's ability to handle large volumes of transactions (spam)
- Implement the above strategy
- Document the findings

## Specification reference

- N/A

## Other information and links

- @ETeissonniere from [Nodle](https://github.com/NodleCode) was gracious enough to open-source [a load-testing utility](https://github.com/NodleCode/chain-loadtest) used by his team.

### PR DESCRIPTION
**Pull request by arijitAD**
## Changes

- Load test submit extrinsic
- Added the logs in this branch https://github.com/ChainSafe/gossamer/commit/782fb1731acf2b9169ad14fa5eef9bf4e154f19d#diff-5076a1f78c8a695913cc20ec45b8751f2fdfad72bc68acbbe7b22761a0d1f0e3

## Tests

Start 3 nodes and fix the authority as Alice, Bob and Charlie in genesis
Genesis:
```
    "Babe": {
                "Authorities": [
                    [
                        "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
                        1
                    ],
                    [
                        "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
                        1
                    ],
                    [
                        "5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y",
                        1
                    ]
                ]
            },
            "Grandpa": {
                "Authorities": [
                    [
                        "5DFNv4Txc4b88qHqQ6GG4D646QcT4fN3jjS2G3r1PyZkfDut",
                        1
                    ],
                    [
                        "5FRbF5E5xpBxaCdjuVkCAtoLgvvoL6wJoE7kAVTUdZBwpgi7",
                        1
                    ],
                    [
                        "5DYo8CvjQcBQFdehVhansDiZCPebpgqvNC8PQPi6K9cL9giT",
                        1
                    ]
                ]
            },
```
Command:
```
Alice
./bin/gossamer --chain gssmr init --force
./bin/gossamer --chain gssmr --key alice --babe-lead --rpc-unsafe-external --ws-unsafe-external --ws

Bob
./bin/gossamer --chain gssmr init --config ~/go/src/github.com/ChainSafe/gossamer/chain/gssmr/config.toml --basepath /tmp/TestSync_ManyProducers/bob --genesis ~/go/src/github.com/ChainSafe/gossamer/chain/gssmr/genesis.json --force
./bin/gossamer --chain gssmr --port 9590 --basepath /tmp/TestSync_ManyProducers/bob --rpchost localhost --rpcport 8541 --rpc --log info --roles 4 --key bob --wsport 9541 --ws --rpc-unsafe-external --ws-unsafe-external

Charlie
./bin/gossamer --chain gssmr init --config ~/go/src/github.com/ChainSafe/gossamer/chain/gssmr/config.toml --basepath /tmp/TestSync_ManyProducers/charlie --genesis ~/go/src/github.com/ChainSafe/gossamer/chain/gssmr/genesis.json --force
./bin/gossamer --chain gssmr --port 9591 --basepath /tmp/TestSync_ManyProducers/charlie --rpchost localhost --rpcport 8542 --rpc --log info --roles 4 --key charlie --wsport 9542 --ws --rpc-unsafe-external --ws-unsafe-external
```

## Issues

- closes #2011 

## Primary Reviewer

- @noot

### CODE REVIEW

File: tests/stress/stress_test.go [hunk 1]
```diff
@@ -569,3 +569,76 @@ func Test_SubmitAndWatchExtrinsic(t *testing.T) {
...
+	require.NoError(t, err)
+
+	var accountInfo types.AccountInfo
+	ok, err = api.RPC.State.GetStorageLatest(key, &accountInfo)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	previous := accountInfo.Data.Free
+	fmt.Printf("%#x has a balance of %v\n", alice, previous)
+	fmt.Printf("You may leave this example running and transfer any value to %#x\n", alice)
```
Comment (qdm12) [inline tests/stress/stress_test.go:615]: If we want to keep logs, change these to `t.Logf` instead?

File: tests/stress/stress_test.go [hunk 2]
```diff
@@ -569,3 +569,76 @@ func Test_SubmitAndWatchExtrinsic(t *testing.T) {
...
+
+	nonce := uint32(accountInfo.Nonce)
+	for i := 0; i < 1000; i++ {
+
```
Comment (qdm12) [inline tests/stress/stress_test.go:619]:
```suggestion
```

File: tests/stress/stress_test.go [hunk 3]
```diff
@@ -569,3 +569,76 @@ func Test_SubmitAndWatchExtrinsic(t *testing.T) {
...
+	meta, err := api.RPC.State.GetMetadataLatest()
+	require.NoError(t, err)
+
+	// Create a call, transferring 12345 units to Bob
+	bob, err := types.NewMultiAddressFromHexAccountID("0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48")
+	require.NoError(t, err)
+
+	// 1 unit of transfer
+	bal, ok := new(big.Int).SetString("1000", 10)
```
Comment (qdm12) [inline tests/stress/stress_test.go:589]: assert `ok` is as expected instead of `err`

**Review by qdm12 — CHANGES_REQUESTED** [review]

File: tests/stress/stress_test.go (hunk 1 above)
Comment (arijitAD) [reply tests/stress/stress_test.go:615]: Done

File: tests/stress/stress_test.go (hunk 2 above)
Comment (arijitAD) [reply tests/stress/stress_test.go:619]: Done

File: tests/stress/stress_test.go (hunk 3 above)
Comment (arijitAD) [reply tests/stress/stress_test.go:589]: Done

**Review by qdm12 — APPROVED** [review]

File: tests/stress/stress_test.go [hunk 4]
```diff
@@ -569,3 +569,75 @@ func Test_SubmitAndWatchExtrinsic(t *testing.T) {
 		`"method":"author_extrinsicUpdate","params":{"result":{"inBlock":"`)
 
 }
+
+func TestSync_SubmitExtrinsicLoad(t *testing.T) {
+	t.Skip()
```
Comment (EclesioMeloJunior) [inline tests/stress/stress_test.go:574]: is this skip intentional?

**Review by EclesioMeloJunior — CHANGES_REQUESTED** [review]

File: tests/stress/stress_test.go (hunk 4 above)
Comment (arijitAD) [reply tests/stress/stress_test.go:574]: Yes, We don't want to load test in CI.

**Pull request merged**